release/release_test.go

```diff
@@ -177,41 +170,56 @@ func (s *ReleaseTestSuite) TestReleaseInfo(c *C) {
 }
 
 func (s *ReleaseTestSuite) TestFilesystemRootType(c *C) {
-	reported_type, err := release.FilesystemRootType()
+	output, err := exec.Command("df", "--output=fstype", "/").CombinedOutput()
 	c.Assert(err, IsNil)
+	// $ df --output=fstype / # Example output on WSL1
+	// Type
+	// wslfs
+
+	fields := strings.Fields(string(output))
+	c.Check(len(fields), Equals, 2)
 
-	// From man stat:
-	// %t   major device type in hex, for character/block device special files
-	output, err := exec.Command("stat", "-f", "-c", "%t", "/").CombinedOutput()
+	wants := strings.TrimSpace(fields[1])
 	c.Assert(err, IsNil)
 
-	outstr := strings.TrimSpace(string(output[:]))
-	statted_type, err := strconv.ParseInt(outstr, 16, 64)
+	got, err := release.FilesystemRootType()
```

Comment:
If we have something like MockProcMounts("some string") that would automatically setup a fake $tmpdir/proc/mounts this is easier to test and does not rely on looking at the host.

Reply:
Done!